### DISCUSSION TITLE
Unexpected useSuspenseQuery behavior with placeholderData

### DISCUSSION THREAD

**Discussion by HJZzzzz** [TanStack/query#6950] [Q&A] [answered]
Thanks for bringing the react-query v5 especially useSuspenseQuery!

When Im upgrading to v5, i noticed that useSuspenseQuery is separate form useQuery, and its said that placeholderData is not applicable to useSuspenseQuery.

However, when i declare my global queryClient like this:

`const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      placeholderData: (previousData) => previousData,
    },
  },
});`

The useSuspenseQuery seems still take the placeholderDataConfig from the global query client.
Its behavior is like a normal useQuery with keepPreviousData. 
1. No fallback will triggered within suspense
2.  UI update wont be suspensed even with startTransition applied.

Understand that it might work if i declare another queryClient for suspense query, but in my project I prefer to only have one queryClient to manage all queries.

And a workaround i found is to manually set the placeholderData to undefined to the useSuspenseQuery.

`  const result = useSuspenseQuery<Res>({
    queryKey: [url, paramsObj],
    queryFn: async () =>
      request<Res>(joinUrlParams(url, params), 'get').catch((error) => {
        console.log(error);
        throw error;
      }),
    ...({ placeholderData: undefined } as any),
    ...options,
  });
`
Is this the correct workaround? May i know if theres any other better approaches?
Thanks!

**Comment by TkDodo** [comment, marked as answer]
right, I didn't really account for having this set globally. I think we need to explicitly set `placeholderData: undefined` in `useSuspenseQuery` here:

https://github.com/TanStack/query/blob/45ef8f8c6bb9a146a41f7ebe72231cc11ccffbe3/packages/react-query/src/useSuspenseQuery.ts#L17-L23

could you PR that ?

**Reply by HJZzzzz** [reply]
sure thing, can i get the permission to push?

**Reply by TkDodo** [reply]
Just create a fork please

**Reply by HJZzzzz** [reply]
https://github.com/TanStack/query/pull/6973
here you go~
